janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -277,28 +273,29 @@ public void restore(Map<String, Map<String, List<IndexEntry>>> documents, KeyInf
             for (final Map.Entry<String, Map<String, List<IndexEntry>>> stores : documents.entrySet()) {
                 final String store = stores.getKey();
                 final IndexWriter writer = getWriter(store, information);
```

Comment:
Not related to this PR but I would like to ask a question. Should we wrap writer into `try catch` and close it if it is opened on failed commit?
I.e. something like:
```
final IndexWriter writer = getWriter(store, information);
try(IndexReader reader = DirectoryReader.open(writer, true, true)){
    // other logic
} finally {
    if (writer.isOpen()){
        writer.close();
    }
}
```

Reply:
Writer is heavy resource and [reused](https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java#L145) for any write operations

If exception occured writer.rollback() will be ok

Reply:
Thank you for the explanation. I missed that it is reused.